test/wallaby/browser_test.exs

```diff
@@ -1,7 +1,122 @@
 defmodule Wallaby.BrowserTest do
-  use ExUnit.Case, async: true
+  use ExUnit.Case
+
+  import Wallaby.SettingsTestHelpers
 
   alias Wallaby.Browser
+  alias Wallaby.Session
+
+  defmodule TestDriver do
+
+    def start_link() do
+      Agent.start_link(fn -> [] end, name: __MODULE__)
```

Comment:
question about this, we register it under the current module name - so this disallows us from running these test async, correct? Couldn't we just start one and embed the information of the pid of the agent in the `TestDriver`/Session struct so that we could still run them in parallel and not rely on the name?

Alternatively, we could rely on module name + current test description (describe + test) if we can get it, should also be unique.

Reply:
Unfortunately we can't run this test case with `async: true` because we're modifying the environment with `Application.put_env(:wallaby, :base_url, "http://example.com")` in some of the tests. 

If that wasn't the case, I agree it would be a great idea to see if we could make this driver not rely on a single registered name. If we reuse this `TestDriver` pattern somewhere else, I'd love to get it to work async.

Reply:
Ah damn, totally I knew I probably overlooked something :D I mean, unless we start putting the base_url in the session right at startup or something like that, but that's a discussion for another time